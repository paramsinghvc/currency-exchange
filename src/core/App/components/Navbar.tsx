import React, { FC } from "react";
import styled from "@emotion/styled";

import theme from "shared/theme";
import Anime from "shared/components/Anime";
import animejs from "animejs";

const Header = styled.header`
  display: flex;
  justify-content: space-between;
`;

const Hamburger = styled.section`
  display: flex;
  flex-direction: column;
  width: 40px;
  margin: 20px;
  > span {
    width: 100%;
    height: 4px;
    background: ${theme.primaryColor};
    margin-bottom: 4px;
    border-radius: 3px;
    &:nth-of-type(2) {
      width: 50%;
    }
    &:last-of-type {
      width: 70%;
      margin-bottom: 0;
    }
  }
`;

const Navbar: FC = () => {
  return (
    <Header>
      <Hamburger role="button">
        <Anime
          open
          duration={800}
          onEntering={{
            opacity: [0, 1],
            translateX: ["-100%", 0],
            scaleX: [1.2],
            delay: animejs.stagger(50),
            easing: "easeInOutQuart"
          }}
          onEntered={{ scaleX: 1, delay: animejs.stagger(75), duration: 300, easing: "linear" }}
        >
          <span></span>
          <span></span>
          <span></span>
        </Anime>
      </Hamburger>
    </Header>
  );
};

export default Navbar;
