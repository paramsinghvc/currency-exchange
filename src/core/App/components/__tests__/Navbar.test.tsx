import React from "react";
import Navbar from "../Navbar";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

const tree = renderer.create(
  <Router>
    <Navbar />
  </Router>
);

it("should render the component", () => {
  expect(tree.toJSON()).toMatchSnapshot();
});
