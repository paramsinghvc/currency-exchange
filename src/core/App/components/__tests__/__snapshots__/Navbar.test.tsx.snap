// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component 1`] = `
<header
  className="css-uf1ume"
>
  <section
    className="css-ei4gva"
    onClick={[Function]}
    role="button"
  >
    <span />
    <span />
    <span />
  </section>
</header>
`;
